client/state/data-layer/wpcom/sites/rewind/credentials/index.js

```diff
@@ -0,0 +1,53 @@
+import { translate } from 'i18n-calypso';
+import {
+	JETPACK_CREDENTIALS_TEST,
+	JETPACK_CREDENTIALS_TEST_FAILURE,
+	JETPACK_CREDENTIALS_TEST_SUCCESS,
+} from 'calypso/state/action-types';
+import { registerHandlers } from 'calypso/state/data-layer/handler-registry';
+import { http } from 'calypso/state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'calypso/state/data-layer/wpcom-http/utils';
+import { errorNotice } from 'calypso/state/notices/actions';
+
+export const testCredentials = ( action ) =>
+	http(
+		{
+			apiNamespace: 'wpcom/v2',
+			method: 'POST',
+			path: `/sites/${ action.siteId }/rewind/credentials/test`,
+			query: {
+				role: action.role,
+			},
+			body: {},
```

Comment:
Is the empty `body` necessary?

Reply:
Unfortunately, it seems there's a bug on `http()` that doesn't consider `apiNamespace` for POST requests if we don't specify `body`. I wasn't able to get where's precisely the bug, but I will double-check it and let you know.

Reply:
If that's the case, that bug sounds like a good opportunity to fix something on the framework side of things, which we're always happy to see.

Reply:
That's a good idea. I'll take notes about it and after this implementation we can consider taking some time and dig a bit for the root cause.

Reply:
Sounds like a good idea for a follow-up 👍